### ISSUE TITLE
Stackless calling for wrappers looks fishy

### ISSUE THREAD

**Issue by akruis** [stackless-dev/stackless#154]
The definition of `STACKLESS_PROMOTE_WRAPPER(descr)` looks suspicious.
```C
#define STACKLESS_PROMOTE_WRAPPER(descr) \
    (slp_try_stackless = stackless & (descr)->d_slpmask) 
```

Why not:
```C
#define STACKLESS_PROMOTE_WRAPPER(descr) \
    (slp_try_stackless = stackless && (descr)->d_slpmask) 
```

The value of variable `stackless`  comes from `slp_try_stackless` (see `STACKLESS_GETARG()`). And the value of `slp_try_stackless` is not very well defined.

By the way, the definition of `STACKLESS_PROMOTE_METHOD()` has the same problem.

**Comment by akruis** [comment]
It is a bug, and the fix is trivial. Just write `&&` instaed of `&`. Fortunately, the bug has no consequences, because it just disables stackless calls in some cases.

Affects all versions.

**Comment by kristjanvalur** [comment]
The original intent was to do a bitwise AND of the two values, if I remember correctly.  d_slpmask contains a bitmask as to what kinds of calls are stackless supported, and the intent was to filter these somehow.  The details were never quite clear to me, though, and it may well have been broken all the time.

**Comment by akruis** [comment]
fixed for 2.7-slp, needs to be ported to 3.x

**Comment by akruis** [comment]
@kristjanvalur Thanks for the info. Probably it was never completely implemented, or it broke during the port to Python 2.

**Comment by akruis** [comment]
Fix ported to master-slp(cfe9e03be4), 3.6-slp (117d9bb461) and 3.5-slp(f13d84199)
